## Descripción del dataset - Enero 2020

El conjunto de datos utilizado corresponde a los viajes en taxi amarillo registrados en la ciudad de Nueva York durante el mes de enero de 2020. Este dataset es proporcionado por el NYC Taxi and Limousine Commission (TLC) y contiene información detallada sobre cada viaje, incluyendo:

- Fecha y hora de inicio y término del viaje
- Distancia recorrida (`trip_distance`)
- Tarifa base (`fare_amount`) y monto total (`total_amount`)
- Propina entregada (`tip_amount`)
- Cantidad de pasajeros (`passenger_count`)
- Ubicación de origen y destino (`PULocationID`, `DOLocationID`)
- Código de tarifa (`RatecodeID`)

Este dataset se utiliza como base para entrenar un modelo de clasificación que predice si un viaje tendrá una propina alta o no, a partir de variables numéricas y categóricas extraídas de esta información.


In [ ]:
import pandas as pd

# Cargar dataset de enero
taxi = pd.read_parquet('https://d37ci6vzurychx.cloudfront.net/trip-data/yellow_tripdata_2020-01.parquet')

In [3]:
# Exploración inicial
taxi.head()

VendorID tpep_pickup_datetime tpep_dropoff_datetime  passenger_count  \
0         1  2020-01-01 00:28:15   2020-01-01 00:33:03              1.0   
1         1  2020-01-01 00:35:39   2020-01-01 00:43:04              1.0   
2         1  2020-01-01 00:47:41   2020-01-01 00:53:52              1.0   
3         1  2020-01-01 00:55:23   2020-01-01 01:00:14              1.0   
4         2  2020-01-01 00:01:58   2020-01-01 00:04:16              1.0   

   trip_distance  RatecodeID store_and_fwd_flag  PULocationID  DOLocationID  \
0            1.2         1.0                  N           238           239   
1            1.2         1.0                  N           239           238   
2            0.6         1.0                  N           238           238   
3            0.8         1.0                  N           238           151   
4            0.0         1.0                  N           193           193   

   payment_type  fare_amount  extra  mta_tax  tip_amount  tolls_amount  \
0             1          6.0    3.0      0.5        1.47           0.0   
1             1          7.0    3.0      0.5        1.50           0.0   
2             1          6.0    3.0      0.5        1.00           0.0   
3             1          5.5    0.5      0.5        1.36           0.0   
4             2          3.5    0.5      0.5        0.00           0.0   

   improvement_surcharge  total_amount  congestion_surcharge airport_fee  
0                    0.3         11.27                   2.5        None  
1                    0.3         12.30                   2.5        None  
2                    0.3         10.80                   2.5        None  
3                    0.3          8.16                   0.0        None  
4                    0.3          4.80                   0.0        None

In [4]:
taxi.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6405008 entries, 0 to 6405007
Data columns (total 19 columns):
 #   Column                 Dtype         
---  ------                 -----         
 0   VendorID               int64         
 1   tpep_pickup_datetime   datetime64[us]
 2   tpep_dropoff_datetime  datetime64[us]
 3   passenger_count        float64       
 4   trip_distance          float64       
 5   RatecodeID             float64       
 6   store_and_fwd_flag     object        
 7   PULocationID           int64         
 8   DOLocationID           int64         
 9   payment_type           int64         
 10  fare_amount            float64       
 11  extra                  float64       
 12  mta_tax                float64       
 13  tip_amount             float64       
 14  tolls_amount           float64       
 15  improvement_surcharge  float64       
 16  total_amount           float64       
 17  congestion_surcharge   float64       
 18  airport_fee           

In [5]:
taxi.describe()

VendorID        tpep_pickup_datetime       tpep_dropoff_datetime  \
count  6.405008e+06                     6405008                     6405008   
mean   1.673002e+00  2020-01-17 03:05:16.413238  2020-01-17 03:21:13.417920   
min    1.000000e+00         2003-01-01 00:07:17         2003-01-01 14:16:59   
25%    1.000000e+00         2020-01-09 17:10:53  2020-01-09 17:27:34.750000   
50%    2.000000e+00         2020-01-16 23:16:29         2020-01-16 23:32:24   
75%    2.000000e+00         2020-01-24 18:24:30         2020-01-24 18:39:51   
max    5.000000e+00         2021-01-02 01:12:10         2021-01-02 01:25:01   
std    4.691265e-01                         NaN                         NaN   

       passenger_count  trip_distance    RatecodeID  PULocationID  \
count     6.339567e+06   6.405008e+06  6.339567e+06  6.405008e+06   
mean      1.515333e+00   2.929644e+00  1.059908e+00  1.647323e+02   
min       0.000000e+00  -3.062000e+01  1.000000e+00  1.000000e+00   
25%       1.000000e+00   9.600000e-01  1.000000e+00  1.320000e+02   
50%       1.000000e+00   1.600000e+00  1.000000e+00  1.620000e+02   
75%       2.000000e+00   2.930000e+00  1.000000e+00  2.340000e+02   
max       9.000000e+00   2.102401e+05  9.900000e+01  2.650000e+02   
std       1.151594e+00   8.315911e+01  8.118432e-01  6.554374e+01   

       DOLocationID  payment_type   fare_amount         extra       mta_tax  \
count  6.405008e+06  6.405008e+06  6.405008e+06  6.405008e+06  6.405008e+06   
mean   1.626627e+02  1.257319e+00  1.269411e+01  1.115456e+00  4.923182e-01   
min    1.000000e+00  0.000000e+00 -1.238000e+03 -2.700000e+01 -5.000000e-01   
25%    1.130000e+02  1.000000e+00  6.500000e+00  0.000000e+00  5.000000e-01   
50%    1.620000e+02  1.000000e+00  9.000000e+00  5.000000e-01  5.000000e-01   
75%    2.340000e+02  2.000000e+00  1.400000e+01  2.500000e+00  5.000000e-01   
max    2.650000e+02  5.000000e+00  4.265000e+03  1.130100e+02  3.080000e+01   
std    6.991261e+01  4.885669e-01  1.212730e+01  1.260054e+00  7.374184e-02   

         tip_amount  tolls_amount  improvement_surcharge  total_amount  \
count  6.405008e+06  6.405008e+06           6.405008e+06  6.405008e+06   
mean   2.189342e+00  3.488395e-01           2.979870e-01  1.866315e+01   
min   -9.100000e+01 -3.574000e+01          -3.000000e-01 -1.242300e+03   
25%    0.000000e+00  0.000000e+00           3.000000e-01  1.116000e+01   
50%    1.950000e+00  0.000000e+00           3.000000e-01  1.430000e+01   
75%    2.860000e+00  0.000000e+00           3.000000e-01  1.980000e+01   
max    1.100000e+03  9.105000e+02           3.000000e-01  4.268300e+03   
std    2.760028e+00  1.766978e+00           3.385937e-02  1.475736e+01   

       congestion_surcharge  
count          6.339567e+06  
mean           2.299052e+00  
min           -2.500000e+00  
25%            2.500000e+00  
50%            2.500000e+00  
75%            2.500000e+00  
max            2.750000e+00  
std            7.017109e-01

## Descripción del Dataset

El diccionario de los datos puede encontrarse [acá](https://www1.nyc.gov/assets/tlc/downloads/pdf/data_dictionary_trip_records_yellow.pdf):

| Field Name      | Description |
| ----------- | ----------- |
| VendorID      | A code indicating the TPEP provider that provided the record. 1= Creative Mobile Technologies, LLC; 2= VeriFone Inc.       |
| tpep_pickup_datetime   | The date and time when the meter was engaged.        |
| tpep_dropoff_datetime   | The date and time when the meter was disengaged.        |
| Passenger_count   | The number of passengers in the vehicle. This is a driver-entered value.      |
| Trip_distance   | The elapsed trip distance in miles reported by the taximeter.      |
| PULocationID   | TLC Taxi Zone in which the taximeter was engaged.      |
| DOLocationID   | TLC Taxi Zone in which the taximeter was disengaged      |
| RateCodeID   | The final rate code in effect at the end of the trip. 1= Standard rate, 2=JFK, 3=Newark, 4=Nassau or Westchester, 5=Negotiated fare, 6=Group ride     |
| Store_and_fwd_flag | This flag indicates whether the trip record was held in vehicle memory before sending to the vendor, aka “store and forward,” because the vehicle did not have a connection to the server. Y= store and forward trip, N= not a store and forward trip |
| Payment_type | A numeric code signifying how the passenger paid for the trip. 1= Credit card, 2= Cash, 3= No charge, 4= Dispute, 5= Unknown, 6= Voided trip |
| Fare_amount | The time-and-distance fare calculated by the meter. |
| Extra | Miscellaneous extras and surcharges. Currently, this only includes the \$0.50 and \$1 rush hour and overnight charges. |
| MTA_tax | \$0.50 MTA tax that is automatically triggered based on the metered rate in use. |
| Improvement_surcharge | \$0.30 improvement surcharge assessed trips at the flag drop. The improvement surcharge began being levied in 2015. |
| Tip_amount | Tip amount – This field is automatically populated for credit card tips. Cash tips are not included. |
| Tolls_amount | Total amount of all tolls paid in trip. |
| Total_amount | The total amount charged to passengers. Does not include cash tips. |